blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -91,16 +101,15 @@ goog.inherits(Blockly.ContractEditor, Blockly.FunctionEditor);
 Blockly.ContractEditor.EXAMPLE_BLOCK_TYPE = 'functional_example';
 Blockly.ContractEditor.EXAMPLE_BLOCK_ACTUAL_INPUT_NAME = 'ACTUAL';
 
-Blockly.ContractEditor.typesToColorsHSV = {
-  // 'none': [0, 0, 0.6], // not available to users
-  'Number': [192, 1.00, 0.99], // 00ccff
-  'string': [180, 1.00, 0.60], // 0099999
-  'image': [285, 1.00, 0.80], // 9900cc
-  'boolean': [90, 1.00, 0.4] // 336600
-};
+Blockly.ContractEditor.typesToColorsHSV = goog.object.filter(
```

Comment:
Ooh, nice catch. We should just iterate over the user choices when presenting to the user & do color lookups on the full set.